Python/specialize.c

```diff
@@ -386,6 +402,13 @@ optimize(SpecializedCacheOrInstruction *quickened, int len)
         if (adaptive_opcode) {
             if (_PyOpcode_InlineCacheEntries[opcode]) {
                 instructions[i] = _Py_MAKECODEUNIT(adaptive_opcode, oparg);
+                if (object_cache_requirements[opcode]) {
+                    assert(_PyOpcode_InlineCacheEntries[opcode] >= 2);
+                    instructions[i+2] = object_offset;
+                    object_offset += object_cache_requirements[opcode];
```

Comment:
This will only ever be `1`, right?

```suggestion
                    object_offset++;
```

Maybe it even makes sense to turn `object_cache_requirements` into an array of `bool`.

Reply:
`bool`s can be expected to take up as much space as a `char`, and maybe we will need this to be 2 for something.
I'd like to have implemented all the specialized instructions before narrowing our options.